osu.Game/Extensions/TaskExtensions.cs

```diff
@@ -13,13 +17,19 @@ public static class TaskExtensions
         /// Avoids unobserved exceptions from being fired.
         /// </summary>
         /// <param name="task">The task.</param>
-        /// <param name="logOnError">Whether errors should be logged as important, or silently ignored.</param>
-        public static void CatchUnobservedExceptions(this Task task, bool logOnError = false)
+        /// <param name="logAsError">
+        /// Whether errors should be logged as errors visible to users, or as debug messages.
+        /// Logging as debug will essentially silence the errors on non-release builds.
+        /// </param>
+        public static void CatchUnobservedExceptions(this Task task, bool logAsError = false)
         {
             task.ContinueWith(t =>
             {
-                if (logOnError)
-                    Logger.Log($"Error running task: {t.Exception?.Message ?? "unknown"}", LoggingTarget.Runtime, LogLevel.Important);
+                Exception? exception = t.Exception?.AsSingular();
+                if (logAsError)
+                    Logger.Error(exception, $"Error running task: {exception?.Message ?? "(unknown)"}", LoggingTarget.Runtime, true);
```

Comment:
Logger.Error is a bit of a special one (submits error logs, but also already includes the message in the output, i think?). I think it may be okay to use here, but have you checked the log output doesn't include duplicated message text?

Reply:
This is how it looks in the log:

```
[runtime] 2020-12-31 11:17:10 [error]: Error running task: Operation is not valid due to the current state of the object.
[runtime] 2020-12-31 11:17:10 [error]: System.InvalidOperationException: Operation is not valid due to the current state of the object.
[runtime] 2020-12-31 11:17:10 [error]: at osu.Game.Online.Multiplayer.MultiplayerClient.StartMatch() in /home/dachb/Dokumenty/opensource/osu/osu.Game/Online/Multiplayer/MultiplayerClient.cs:line 178
```

and on screen:

![osu_2020-12-31_12-17-40](https://user-images.githubusercontent.com/20418176/103408337-56cbb100-4b62-11eb-8329-8d6d80ee0330.jpg)

I guess the message is duplicated but I mostly included it, because without it the message looks barren/jank (just an "Error running task"). I guess that can be changed by adjusting the error messages but not sure if worth?

Reply:
If I were to suggest an alternative it would be to include a string param to `CatchUnobservedExceptions()` and adjust usages to provide context as to which operation is being bailed out of.

Reply:
I think it's fine, as long a the text makes it easy to know the source is a call to this method.